app/org/sagebionetworks/bridge/services/email/BasicEmailProvider.java

```diff
@@ -0,0 +1,106 @@
+package org.sagebionetworks.bridge.services.email;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeBodyPart;
+
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.config.BridgeConfigFactory;
+import org.sagebionetworks.bridge.models.studies.EmailTemplate;
+import org.sagebionetworks.bridge.models.studies.Study;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+public class BasicEmailProvider implements MimeTypeEmailProvider {
+
+    private final Study study;
+    private final Set<String> recipientEmails;
+    private final Map<String,String> tokenMap;
+    private final EmailTemplate template;
+    
+    private BasicEmailProvider(Study study, Map<String,String> tokenMap, Set<String> recipientEmails, EmailTemplate template) {
+        this.study = study;
+        this.recipientEmails = recipientEmails;
+        this.tokenMap = tokenMap;
+        this.template = template;
+    }
+    public Study getStudy() {
+        return study;
+    }
+    public Set<String> getRecipientEmails() {
+        return recipientEmails;
+    }
+    public Map<String,String> getTokenMap() {
+        return ImmutableMap.copyOf(tokenMap);
+    }
+    public EmailTemplate getTemplate() {
+        return template;
+    }
+    
+    @Override
+    public MimeTypeEmail getMimeTypeEmail() throws MessagingException {
+        tokenMap.put("studyName", study.getName());
+        tokenMap.put("studyId", study.getIdentifier());
+        tokenMap.put("supportEmail", study.getSupportEmail());
+        tokenMap.put("technicalEmail", study.getTechnicalEmail());
+        tokenMap.put("sponsorName", study.getSponsorName());
+        tokenMap.put("host", BridgeConfigFactory.getConfig().getHostnameWithPostfix("webservices"));
+        
+        final MimeTypeEmailBuilder builder = new MimeTypeEmailBuilder();
+
+        final String formattedSubject = BridgeUtils.resolveTemplate(template.getSubject(), tokenMap);
+        builder.withSubject(formattedSubject);
+
+        Set<String> senderEmails = BridgeUtils.commaListToOrderedSet(study.getSupportEmail());
```

Comment:
If they create a "public" study that uses these workflows, they have to verify that email address, there are no exceptions. If you want to automate enforcement of this somehow (beyond the fact that people get an email telling them to verify the email address from SES), please file a JIRA. We could show an error when they sign in to the Bridge Study Manager, for example.